src/osd/OSD.cc

```diff
@@ -8308,63 +8311,72 @@ void OSD::handle_osd_map(MOSDMap *m)
 }
 
 /*
- *  For each added map which was sent in this MOSDMap [first, last]:
- *
- *  1) Check if a pool was deleted
- *  2) For existing pools, check if pg_num was changed
- *  3) Check if a pool was created
- *
- *  Track all of the changes above in pg_num_history
+ *  Compare between the previous last_map to the current one
+ *  Track all of the changes relevant in pg_num_history
  */
 void OSD::track_pools_and_pg_num_changes(map<epoch_t,OSDMapRef>& added_maps,
                                          ObjectStore::Transaction& t,
+                                         epoch_t first,
```

Comment:
Why do we need to pass first and last?  Shouldn't they be the first and last entries in added_maps in all cases?

Reply:
added_maps should be const.

Reply:
> Why do we need to pass first and last? Shouldn't they be the first and last entries in added_maps in all cases?

I find it easier to read/use `first` and `last` rather then begin/rbegin. No other actual benefits.

Reply:
I don't like that it introduces a way to pass incorrect values for first and last.  If you are going to allow them to be passed independently, you need function documentation indicating, at least, that added_maps must be a superset all of the maps in [first, last].